File: client/golang/prsim/evaluator.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package prsim

import (
	"context"
	"github.com/be-io/mesh/client/golang/types"
)

var IEvaluator = (*Evaluator)(nil)

// Evaluator
// @MPI(macro.MeshMPI)
// @SPI(macro.MeshSPI)
type Evaluator interface {

	// Compile the named rule.
	// @MPI("mesh.eval.compile")
	Compile(ctx context.Context, script *types.Script) (string, error)

	// Exec the script with name.
	// @MPI("mesh.eval.exec")
	Exec(ctx context.Context, code string, args map[string]string, dft string) (string, error)

	// Dump the scripts.
	// @MPI("mesh.eval.dump")
	Dump(ctx context.Context, feature map[string]string) ([]*types.Script, error)

	// Index the scripts.
	// @MPI("mesh.eval.index")
	Index(ctx context.Context, index *types.Paging) (*types.Page[*types.Script], error)
}

File: client/javascript/src/prsim/evaluator.ts
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

import {spi} from "@/macro/spi";
import {mpi} from "@/macro/mpi";
import type {Context} from "@/prsim/context";
import {Status} from "@/cause";
import {Page, Paging, Script} from "@/kinds";
import {index} from "@/macro";


@spi("mesh")
export abstract class Evaluator {

    /**
     * Compile the named rule.
     */
    @mpi("mesh.eval.compile", String)
    compile(ctx: Context, @index(0, 'script') script: Script): Promise<string> {
        return Promise.reject(Status.URN_NOT_PERMIT)
    }

    /**
     * Exec the script with name.
     */
    @mpi("mesh.eval.exec", String)
    exec(ctx: Context, @index(0, 'code') code: string, @index(1, 'args') args: Map<string, string>, @index(2, 'dft') dft: string): Promise<string> {
        return Promise.reject(Status.URN_NOT_PERMIT)
    }

    /**
     * Dump the scripts.
     */
    @mpi("mesh.eval.dump", [Array, Script])
    dump(ctx: Context, @index(0, 'feature') feature: Map<string, string>): Promise<Script[]> {
        return Promise.reject(Status.URN_NOT_PERMIT)
    }

    /**
     * Index the scripts.
     */
    @mpi("mesh.eval.index", [Page, Script])
    index(ctx: Context, @index(0, 'index') index: Paging): Promise<Page<Script>> {
        return Promise.reject(Status.URN_NOT_PERMIT)
    }

}